matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/threads/DefaultThreadsService.kt

```diff
@@ -17,48 +17,47 @@
 package org.matrix.android.sdk.internal.session.room.threads
 
 import androidx.lifecycle.LiveData
+import androidx.paging.PagedList
 import com.zhuinden.monarchy.Monarchy
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
 import io.realm.Realm
+import kotlinx.coroutines.CoroutineScope
 import org.matrix.android.sdk.api.session.room.threads.ThreadsService
 import org.matrix.android.sdk.api.session.room.threads.model.ThreadSummary
 import org.matrix.android.sdk.internal.database.helper.enhanceWithEditions
 import org.matrix.android.sdk.internal.database.helper.findAllThreadsForRoomId
 import org.matrix.android.sdk.internal.database.mapper.ThreadSummaryMapper
-import org.matrix.android.sdk.internal.database.mapper.TimelineEventMapper
 import org.matrix.android.sdk.internal.database.model.threads.ThreadSummaryEntity
 import org.matrix.android.sdk.internal.di.SessionDatabase
-import org.matrix.android.sdk.internal.di.UserId
 import org.matrix.android.sdk.internal.session.room.relation.threads.FetchThreadSummariesTask
 import org.matrix.android.sdk.internal.session.room.relation.threads.FetchThreadTimelineTask
 
 internal class DefaultThreadsService @AssistedInject constructor(
         @Assisted private val roomId: String,
-        @UserId private val userId: String,
         private val fetchThreadTimelineTask: FetchThreadTimelineTask,
         private val fetchThreadSummariesTask: FetchThreadSummariesTask,
         @SessionDatabase private val monarchy: Monarchy,
-        private val timelineEventMapper: TimelineEventMapper,
-        private val threadSummaryMapper: ThreadSummaryMapper
+        private val threadSummaryMapper: ThreadSummaryMapper,
 ) : ThreadsService {
 
     @AssistedFactory
     interface Factory {
         fun create(roomId: String): DefaultThreadsService
     }
 
-    override fun getAllThreadSummariesLive(): LiveData<List<ThreadSummary>> {
-        return monarchy.findAllMappedWithChanges(
-                { ThreadSummaryEntity.findAllThreadsForRoomId(it, roomId = roomId) },
-                {
-                    threadSummaryMapper.map(it)
-                }
+    override suspend fun getPagedThreadsList(coroutineScope: CoroutineScope, userParticipating: Boolean): LiveData<PagedList<ThreadSummary>> {
+        return DefaultGetPagedThreadListTask(monarchy, threadSummaryMapper, fetchThreadSummariesTask).execute(
```

Comment:
You should inject the task `GetPagedThreadListTask` in the constructor instead of creating a Default instance here.

Reply:
That's actually a thing i'd like to discuss. This task is basically a single-use cache, so we need to ensure that if the same task is called again, e.g. when filters are changed, it won't use anything from the previous run. If I'd inject the task it will require to do some cleanup on every execute and could cause problems if anything is not cleared, while running it from the scratch ensure that no previous data remain. Do you think it will be better to inject and do a cleanup?

Reply:
A Task shouldn't keep data, it's supposed to be a one/off operation